helpers.lua

```diff
@@ -35,10 +37,44 @@ local scroller = {}
 -- }}}
 
 -- {{{ Helper functions
+-- {{{ Determine operating system
+function helpers.getos()
+    local f = io.popen("uname -s")
+    local uname = f:read("*line")
+    f:close()
+
+    return uname:lower()
+end
+-- }}}
+
 -- {{{ Loader of vicious modules
 function helpers.wrequire(table, key)
-    local module = rawget(table, key)
-    return module or require(table._NAME .. "." .. key)
+    local ret = rawget(table, key)
+
+    if not ret then
+        os = helpers.getos()
```

Comment:
If I got it right, this code should only be called once for every widget (at creation time), so I think it isn't such an overhead. But using a cached value is an option, too!

The other commits change this function a bit to solve the problem more elegant.